### DISCUSSION TITLE
Cannot get Tenant Service Principal to Work

### DISCUSSION THREAD

**Discussion by SP-Support** [YannickRe/spfx-msgraph-peoplesearch#59] [Q&A] [answered]
When I run the `Grant-PnPTenantServicePrincipalPermission -Scope "User.Read.All" -Resource 'Microsoft Graph'` I get  the error Grant-PnPTenantServicePrincipalPermission : Access Token missing or malformed.

I am not sure how to get this app to recognize it has Graph Read access.

**Comment by YannickRe** [comment, marked as answer]
How did you connect?
I suggest you do like this:
```
Connect-PnPOnline https://{tenantName}-admin.sharepoint.com -Interactive
Grant-PnPTenantServicePrincipalPermission -Scope "User.Read.All" -Resource 'Microsoft Graph'
```

**Reply by SP-Support** [reply]
I had tried that. I discovered I had a bigger issue and had more than one service principal. We recently migrated tenants and we are still cleaning you.  The app works GREAT. Thanks for your contribution to the community.

**Reply by YannickRe** [reply]
Happy to hear you figured it out, and that you like the app! Let me know if there is something I can do for you :)
